bin/run.py

```diff
@@ -220,7 +226,17 @@ def _validate_output(self, bar):
 class Submission(program.Program):
     def __init__(self, problem, path, skip_double_build_warning=False):
         super().__init__(
-            problem, path, 'submissions', skip_double_build_warning=skip_double_build_warning
+            problem,
+            path,
+            'submissions',
+            limits={
+                'code': problem.limits.code,
+                'compilation_time': problem.limits.compilation_time,
+                'compilation_memory': problem.limits.compilation_memory,
+                'timeout': problem.settings.timeout,
```

Comment:
hmm; if we merged this into problem.limits, or else made it a separate argument, you could just pass problem.limits as-is here.

Also since you do use `self.limits['timeout']` now, but not the others.
Other thoughts:
- could make limits into a NameSpace object so `.timeout` works.
- Could probably do some dict-merge to avoid spelling all the fields.

Edit: saw the other comment that each type of program infers it's own correct limits, so most of this is probably moot.

Reply:
Yeah, there was also one thread where I was confused with `Problem.limits` vs. `Program.limits` :stuck_out_tongue: https://github.com/RagnarGrootKoerkamp/BAPCtools/pull/415#discussion_r1938240592